### PR TITLE
Update user agent to use product version instead of assembly version

### PR DESCRIPTION
**Pull request by abhilasharora**
This pull request updates the user agent string to include the package version (informational version) instead of the assembly version. Check out [this blog](https://andrewlock.net/version-vs-versionsuffix-vs-packageversion-what-do-they-all-mean/) for more details on the different versions in .NET.

### CODE REVIEW

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs [hunk 1]
```diff
@@ -23,14 +22,13 @@ internal static class TracingUtils
         public static string GenerateUserAgent(string currentUserAgent = null)
         {
             Assembly assembly = typeof(AzureAppConfigurationOptions).Assembly;
-            var userAgent = new StringBuilder($"{assembly.GetName().Name}/{assembly.GetName().Version}");
+            string informationalVersion = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
```
Comment (zhenlan) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs]: Just to understand, who will set the `AssemblyInformationalVersionAttribute`?

File: tests/Tests.AzureAppConfiguration/Tests.cs [hunk 2]
```diff
@@ -208,6 +208,10 @@ public void TestUserAgentHeader()
             MockRequest request = mockTransport.SingleRequest;
 
             string appUserAgent = TracingUtils.GenerateUserAgent("SdkUserAgent");
+
+            // Validate the user agent has information version (3.9.9999) instead of assembly version (3.9.9999.0)
+            Assert.Equal("Microsoft.Extensions.Configuration.AzureAppConfiguration/3.9.9999 SdkUserAgent", appUserAgent);
```
Comment (zhenlan) [inline tests/Tests.AzureAppConfiguration/Tests.cs]: I am wondering if this test will pass in an official build.

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs (hunk 1 above)
Comment (abhilasharora) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs]: This attribute is automatically generated by the `.NET Core SDK` based on the `Version` property in the `csproj` file. When I build the project, I can see the `*AssemblyInfo.cs` file in the obj directory with the `AssemblyInformationalVersionAttribute` attribute along with other auto-generated attributes, and these are also applied to the corresponding assembly. More information can be found [here](https://compiledexperience.com/blog/posts/assemblyinfo-generation).

Comment (zhenlan) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs]: Thanks.

**Pull request merged**